core/src/main/java/io/grpc/Metadata.java

```diff
@@ -68,8 +68,9 @@
   /**
    * Simple metadata marshaller that encodes strings as is.
    *
-   * <p>This should be used with ASCII strings that only contain printable characters and space.
-   * Otherwise the output may be considered invalid and discarded by the transport.
+   * <p>This should be used with ASCII strings that only contain the characters listed in the class
+   * comment of {@link AsciiMarshaller}. Otherwise the output may be considered invalid and
+   * discarded by the transport.
```

Comment:
I think we should reserve the right to fail the call as well. It may be discarded or the call may fail.